src/control/server/server_utils.go

```diff
@@ -264,21 +288,22 @@ func initFirstEngine(engine *EngineInstance, sysdb *system.Database, joinFn syst
 		}
 		return joinFn(ctx, req)
 	}
-
-	return nil
 }
 
-func exportStats(ctx context.Context, allStarted *sync.WaitGroup, log *logging.LeveledLogger, port int, instances []*EngineInstance) {
-	allStarted.Wait()
-
-	if port == 0 {
+// registerTelemetryCallback sets callback which will launch Prometheus
```

Comment:
nit: I would name this `registerTelemetryCallbacks()` and omit the reference to Prometheus in the comment. We will likely use this as a place to integrate other telemetry target as well (e.g. InfluxDB, etc).